clients/fides-js/src/fides-ext-gpp.ts

```diff
@@ -41,39 +45,87 @@ declare global {
 }
 
 /**
- * Wrapper around setting a TC string on the CMP API object
+ * Wrapper around setting a TC string on the CMP API object.
+ * Returns whether or not the TC string was set.
  * @param event: FidesEvent
  * @param cmpApi: the CMP API model
  */
 const setTcString = (event: FidesEvent, cmpApi: CmpApi) => {
+  if (!window.Fides.options.tcfEnabled) {
+    return false;
+  }
   const tcString = extractTCStringForCmpApi(event);
-  if (tcString) {
-    // Workaround for bug in base library https://github.com/IABTechLab/iabgpp-es/issues/35
-    cmpApi.setFieldValueBySectionId(TCF_SECTION_ID, "CmpId", ETHYCA_CMP_ID);
-    cmpApi.setSectionStringById(TCF_SECTION_ID, tcString);
+  if (!tcString) {
+    return false;
+  }
+  // Workaround for bug in base library https://github.com/IABTechLab/iabgpp-es/issues/35
+  cmpApi.setFieldValueBySectionId(TcfEuV2.ID, "CmpId", ETHYCA_CMP_ID);
+  cmpApi.setSectionStringById(TcfEuV2.ID, tcString);
+  return true;
+};
+
+/** From our options, derive what APIs of GPP are applicable */
+const getSupportedApis = () => {
+  const supportedApis: string[] = [];
+  // QUESTION: or should this come from GPP config?
+  if (window.Fides.options.tcfEnabled) {
+    supportedApis.push(`${TcfEuV2.ID}:${TcfEuV2.NAME}`);
+    return supportedApis;
+  }
+  if (isPrivacyExperience(window.Fides.experience)) {
+    const { gpp_settings: gppSettings } = window.Fides.experience;
+    if (gppSettings && gppSettings.enabled && gppSettings.regions) {
+      const gppSections = Object.values(FIDES_REGION_TO_GPP_SECTION);
+      gppSettings.regions.forEach((region) => {
+        const section = gppSections.find((d) => d.prefix === region);
+        if (section) {
+          supportedApis.push(`${section.id}:${section.prefix}`);
+        }
+      });
+    }
   }
+  return supportedApis;
 };
 
 export const initializeGppCmpApi = () => {
   makeStub();
   const cmpApi = new CmpApi(ETHYCA_CMP_ID, CMP_VERSION);
-  cmpApi.setApplicableSections([TCF_SECTION_ID]);
   cmpApi.setCmpStatus(CmpStatus.LOADED);
   // If consent does not need to be resurfaced, then we can set the signal to Ready here
   window.addEventListener("FidesInitialized", (event) => {
     const { experience } = window.Fides;
+    cmpApi.setSupportedAPIs(getSupportedApis());
     if (
       isPrivacyExperience(experience) &&
       !shouldResurfaceConsent(experience, event.detail)
     ) {
-      setTcString(event, cmpApi);
+      const tcSet = setTcString(event, cmpApi);
+      if (tcSet) {
+        cmpApi.setApplicableSections([TcfEuV2.ID]);
+      }
+      setGppNoticesProvidedFromExperience({ cmpApi, experience });
```

Comment:
this is the case where a user returns and already has a cookie, so we need to initialize the CMP with the right data. we set the opt outs, and I assume we should also set the `*Notices` but I'd like that assumption checked @mfbrown 🙏

Reply:
I think it's fair to say that if a user returns with an existing opt-out value, then we can report that they've also seen the notice in the GPP string.